src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -120,51 +123,42 @@
  * If the value to be read is stored in memory using {@linkplain ByteOrder#BIG_ENDIAN big-endian} encoding, the access operation
  * can be expressed as follows:
  * {@snippet lang=java :
- * MemorySegment segment = ...
  * int value = segment.get(ValueLayout.JAVA_INT.withOrder(BIG_ENDIAN), 0);
  * }
  *
- * More complex access operations can be implemented using var handles. The {@link ValueLayout#varHandle()}
- * method can be used to obtain a var handle that can be used to get/set values represented by the given value layout on a memory segment.
- * A var handle obtained from a layout supports several additional <a href=MemoryLayout.html#access-mode-restrictions>
- * access modes</a>. More importantly, var handles can be <em>combined</em> with method handles to express complex access
- * operations. For instance, a var handle that can be used to access an element of an {@code int} array at a given logical
- * index can be created as follows:
+ * Access operations on memory segments are implemented using var handles. The {@link ValueLayout#varHandle()}
+ * method can be used to obtain a var handle that can be used to get/set values represented by the given value layout
+ * on a memory segment at the given offset:
  *
  * {@snippet lang=java:
- * MemorySegment segment = ...
- * VarHandle intHandle = ValueLayout.JAVA_INT.varHandle(); // (MemorySegment, long)
- * MethodHandle scale = ValueLayout.JAVA_INT.scaleHandle(); // <base offset> + <index> * JAVA_INT.byteSize()
+ * VarHandle intAtOffsetHandle = ValueLayout.JAVA_INT.varHandle(); // (MemorySegment, long)
+ * int value = (int) intAtOffsetHandle.get(segment, 10L);          // segment.get(ValueLayout.JAVA_INT, 10L)
+ * }
+ *
+ * The var handle returned by {@link ValueLayout#varHandle()} features a <em>base offset</em> parameter. This parameter
+ * allows clients to express complex access operations, by injecting additional offset computation into the var handle.
+ * For instance, a var handle that can be used to access an element of an {@code int} array at a given logical
+ * index can be created as follows:
  *
- * intHandle = MethodHandles.filterCoordinates(intHandle, 1, scale);
- * int value = (int) intHandle.get(segment, 0L, 3L); // get int element at offset 0 + 3 * 4 = 12
+ * {@snippet lang=java:
+ * MethodHandle scale = ValueLayout.JAVA_INT.scaleHandle();              // (long, long)long
+ * VarHandle intAtOffsetAndIndexHandle =
+ *         MethodHandles.filterCoordinates(intAtOffsetHandle, 1, scale); // (MemorySegment, long, long)
```

Comment:
This should be `collectCoordinates`. (`filterCoordinates` only works for filters with a single parameter)